sift/src/test/java/siftscience/android/SiftTest.java

```diff
@@ -581,6 +581,54 @@ public void testAppStateCollectorWithMockLocation() {
         assertEquals(mockLocationJson, event.androidAppState.location);
     }
 
+    @Test
+    public void testAppStateForceUpload() {
+        MemorySharedPreferences preferences = new MemorySharedPreferences();
+
+        Sift.Config config = new Sift.Config.Builder().withDisallowLocationCollection(true).build();
+
+        SiftImpl sift = new SiftImpl(mockContext(preferences), config, "",
+                false, mockTaskManager());
+        assertTrue(preferences.fields.isEmpty());
+
+        AppStateCollector appStateCollector = new AppStateCollector(sift, mockContext(preferences));
+        appStateCollector.collect();
+        // Collect twice because the first one gets uploaded and flushed
+        appStateCollector.collect();
+        sift.save();
+
+        sift.forceUploadAppStateEvent();
+        // Force upload will upload the event and flush the queue
+
+        List<MobileEventJson> eventList = sift.getQueue(SiftImpl.APP_STATE_QUEUE_IDENTIFIER).flush();
+        assertEquals(0, eventList.size());
+
+    }
+
+    @Test
+    public void testDevicePropertyForceUpload() {
+        MemorySharedPreferences preferences = new MemorySharedPreferences();
+
+        Sift.Config config = new Sift.Config.Builder().build();
+
+        SiftImpl sift = new SiftImpl(mockContext(preferences), config, "",
+                false, mockTaskManager());
+        assertTrue(preferences.fields.isEmpty());
+
+        DevicePropertiesCollector devicePropertiesCollector = new DevicePropertiesCollector(sift, mockContext(preferences));
+        devicePropertiesCollector.collect();
+        // First one gets uploaded and flushed
+
+        devicePropertiesCollector.collect();
```

Comment:
I am not sure why the double collect is required here as well. I suggest just having the single collect call. I understand that the upload will not do anything because the collect will upload the device property, but if we changed the queue configuration to 2 then we would expect that the collect would gather one device property and upload would force upload that device property, right?

Reply:
Single collect will not keep the event in the queue as per the `isReadyForUpload()` condition check. So, to test the device property force upload for configuration > 1, we need to call double collect to make sure that there is a single event in the queue and it will flush after upload.